sdks/python/apache_beam/runners/portability/portable_runner_test.py

```diff
@@ -300,6 +301,42 @@ def test__create_process_environment(self):
                 command='run.sh',
             ).SerializeToString()))
 
+  def test__create_external_environment(self):
+    self.assertEqual(
+        PortableRunner._create_environment(PipelineOptions.from_dictionary({
+            'environment_type': "EXTERNAL",
+            'environment_config': 'localhost:50000',
+        })), beam_runner_api_pb2.Environment(
+            urn=common_urns.environments.EXTERNAL.urn,
+            payload=beam_runner_api_pb2.ExternalPayload(
+                endpoint=endpoints_pb2.ApiServiceDescriptor(
+                    url='localhost:50000')
+            ).SerializeToString()))
+    self.assertEqual(
+        PortableRunner._create_environment(PipelineOptions.from_dictionary({
+            'environment_type': "EXTERNAL",
+            'environment_config': ' {"url":"localhost:50000", '
```

Comment:
Should we also be testing the case without any space at the beginning?

Reply:
I updated the test for this part in this commit: https://github.com/apache/beam/pull/9730/commits/916e170efb2e8f2ec80dba10bc26f8bf00467988